crypto-bigint/src/uint/shl.rs

```diff
@@ -21,17 +22,20 @@ impl<const LIMBS: usize> UInt<LIMBS> {
         }
 
         let shift_num = n / BIT_SIZE;
-        let shift_rem = n % BIT_SIZE;
+        let rem = n % BIT_SIZE;
+        let nz = is_nonzero(rem as Inner);
+        let lshift_rem = rem as Inner;
+        let rshift_rem = ct_select(0, (BIT_SIZE - rem) as Inner, nz);
```

Comment:
It seems like you could use `subtle` for this:

```suggestion
        let lshift_rem = rem as Inner;
        let rshift_rem = Inner::conditional_select(0, (BIT_SIZE - n) as Inner), !rem.ct_eq(&0));
```

Reply:
Subtle is not a const fn

Reply:
In order to have const fn for shift you have to use these. I would prefer subtle than reinventing the wheel

Reply:
Once traits support const fn in stable then we can remove these

Reply:
Aha, makes sense